frame/treasury/src/lib.rs

```diff
@@ -204,6 +243,33 @@ pub mod pallet {
 		/// process. The `Success` value is the maximum amount that this origin is allowed to
 		/// spend at a time.
 		type SpendOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = BalanceOf<Self, I>>;
+
+		/// Type parameter representing the asset kinds to be spent from the treasury.
+		type AssetKind: Parameter + MaxEncodedLen;
+
+		/// Type parameter used to identify the beneficiaries eligible to receive treasury spend.
+		type Beneficiary: Parameter + MaxEncodedLen;
+
+		/// Converting trait to take a source type and convert to [`Self::Beneficiary`].
+		type BeneficiaryLookup: StaticLookup<Target = Self::Beneficiary>;
+
+		/// Type for processing spends of [Self::AssetKind] in favor of [`Self::Beneficiary`].
+		type Paymaster: Pay<Beneficiary = Self::Beneficiary, AssetKind = Self::AssetKind>;
+
+		/// Type to convert the balance of an [`Self::AssetKind`] to balance of the native asset.
+		type BalanceConverter: ConversionFromAssetBalance<
+			<Self::Paymaster as Pay>::Balance,
+			Self::AssetKind,
+			BalanceOf<Self, I>,
+		>;
```

Comment:
Perhaps something that should be clarified in docs: Within the pallet this does not actually convert the asset (or native asset) but is only used for ensuring that proposed amount respects the origin limitations. The actual payout assumes that the Treasury has the asset it is supposed to spend, but does not go about acquiring it.

Reply:
updated